ntc_templates/templates/huawei_smartax_display_ont_wan-info_0_1_2_0.textfsm

```diff
@@ -0,0 +1,80 @@
+Value INDEX (\d+)
+Value NAME (\S+|\S.+\S)
+Value SERVICE_TYPE (\S+)
+Value CONNECTION_TYPE (\S.+\S)
+Value IPV4_CONNECTION_STATUS (\S+)
+Value IPV4_ACCESS_TYPE (\S+)
+Value IPV4_ADDRESS (\S+)
+Value SUBNET_MASK (\S+)
+Value DEFAULT_GATEWAY (\S+)
+Value PRIMARY_DNS (\S+)
+Value SECONDARY_DNS (\S+)
+Value MANAGE_VLAN (\d+)
+Value MANAGE_PRIORITY (\d+)
+Value MULTICAST_VLAN (\S+)
+Value NAT_SWITCH (\S+)
+Value OPTION60 (\S+)
+Value SWITCH (\S+)
+Value MAC_ADDRESS (\S+)
+Value PRIORITY_POLICY (\S+)
+Value L2_ENCAP_TYPE (\S+)
+Value IPV4_SWITCH (\S+)
+Value IPV6_SWITCH (\S+)
+Value PREFIX (\S+)
+Value PREFIX_ACCESS_MODE (\S+)
+Value PREFIX_PREFERRED_TIME (\S+)
+Value PREFIX_VALID_TIME (\S+)
+Value IPV6_CONNECTION_STATUS (\S+)
+Value IPV6_ADDRESS (\S+)
+Value IPV6_PRIMARY_DNS (\S+)
+Value IPV6_SECONDARY_DNS (\S+)
+Value IPV6_MULTICAST_VLAN (\S+)
+Value IPV6_ADDRESS_STATUS (\S+)
+Value IPV6_ADDRESS_ACCESS_MODE (\S+)
+Value IPV6_ADDRESS_PREFERRED_TIME (\S+)
+Value IPV6_ADDRESS_VALID_TIME (\S+)
+Value DS_LITE_MODE (\S+)
+Value DS_LITE_PEER_ADDRESS (\S+)
+
+Start
+  ^\s*-+\s*$$
+  ^\s*F\/S\/P\s*:\s+(\S+)\s*$$
+  ^\s*ONT ID\s*:\s+(\S+)\s*$$
+  ^\s*Index\s*:\s+(-|${INDEX})\s*$$
+  ^\s*Name\s*:\s+(-|${NAME})\s*$$
+  ^\s*Service\s+type\s*:\s+(-|${SERVICE_TYPE})\s*$$
+  ^\s*Connection\s+type\s*:\s+(-|${CONNECTION_TYPE})\s*$$
+  ^\s*IPv4\s+Connection\s+status\s*:\s+(-|${IPV4_CONNECTION_STATUS})\s*$$
+  ^\s*IPv4\s+access\s+type\s*:\s+(-|${IPV4_ACCESS_TYPE})\s*$$
+  ^\s*IPv4\s+address\s*:\s+(-|${IPV4_ADDRESS})\s*$$
+  ^\s*Subnet\s+mask\s*:\s+(-|${SUBNET_MASK})\s*$$
+  ^\s*Default\s+gateway\s*:\s+(-|${DEFAULT_GATEWAY})\s*$$
+  ^\s*Primary\s+DNS\s*:\s+(-|${PRIMARY_DNS})\s*$$
+  ^\s*Secondary\s+DNS\s*:\s+(-|${SECONDARY_DNS})\s*$$
+  ^\s*Manage\s+VLAN\s*:\s+(-|${MANAGE_VLAN})\s*$$
+  ^\s*Manage\s+priority\s*:\s+(-|${MANAGE_PRIORITY})\s*$$
+  ^\s*Multicast\s+VLAN\s*:\s+(-|${MULTICAST_VLAN})\s*$$
+  ^\s*NAT\s+switch\s*:\s+(-|${NAT_SWITCH})\s*$$
+  ^\s*Option60\s*:\s+(-|${OPTION60})\s*$$
+  ^\s*Switch\s*:\s+(-|${SWITCH})\s*$$
+  ^\s*MAC\s+address\s*:\s+(--------------|${MAC_ADDRESS})\s*$$
```

Comment:
I'd simplify the missing MAC address pattern with a regex just in case the developers ever add or remove a dash.

```suggestion
  ^\s*MAC\s+address\s*:\s+(-+|${MAC_ADDRESS})\s*$$
```

Reply:
@evilmonkey19
See this note/suggestion?